### PR TITLE
Don't disable other container engines when --vm_driver=none

### LINKED ISSUE #4541: NoneDriver: Minikube start restarts containerd, hence it kills all my non k8s containers
I am using version minikube 1.1.1

Whenever I do a 

`sudo -E minikube start --vm-driver=none --extra-config kubeadm.ignore-preflight-errors=SystemVerification ${EXTRA_DNS_PARAMETER}  --logtostderr`

I see this:

```
I0620 13:33:02.265561   22899 exec_runner.go:39] Run: systemctl is-active --quiet service containerd
I0620 13:33:02.279081   22899 exec_runner.go:39] Run: sudo systemctl stop containerd
I0620 13:33:02.361533   22899 exec_runner.go:39] Run: systemctl is-active --quiet service containerd
I0620 13:33:02.381293   22899 exec_runner.go:39] Run: systemctl is-active --quiet service crio
I0620 13:33:02.396272   22899 exec_runner.go:39] Run: sudo systemctl start docker
```

which I traced to 

https://github.com/kubernetes/minikube/blob/master/pkg/minikube/cruntime/docker.go#L78

```
// Enable idempotently enables Docker on a host
func (r *Docker) Enable() error {
	if err := disableOthers(r, r.Runner); err != nil {
		glog.Warningf("disableOthers: %v", err)
	}
	return r.Runner.Run("sudo systemctl start docker")
}

```

`disableOthers` is here by the way: https://github.com/kubernetes/minikube/blob/master/pkg/minikube/cruntime/cruntime.go#L96

What's the rationale for that ?
Why does minikube bother messing with my container daemons in first place ?

Should we just remove  `disableOthers` ? Should we remove it only for the none driver ?
Should we just not restart `containerd` if docker is > SOME_VERSION ?

Should we just not restart `containerd` because today everybody uses it, hence it shouldn't just be restarted ?

All the solutions are trivial to implement. Hence I am opening the discussion so I can create the right PR :)

### PR DESCRIPTION
**Pull request by marcosdiez**
closes #4541 

it speedups a second on two the minikube launch here !

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -672,7 +672,8 @@ func configureRuntimes(runner cruntime.CommandRunner) cruntime.Manager {
 		exit.WithError(fmt.Sprintf("Failed runtime for %+v", config), err)
 	}
 
-	err = cr.Enable()
+	disableOtherContainerEngines := viper.GetString(vmDriver) != constants.DriverNone
```
Comment (tstromberg) [inline cmd/minikube/cmd/start.go]: While this is clever, let's aim for easier readability:

```
disableOthers := true
if  viper.GetString(vmDriver) == constants.DriverNone {
  disableOthers = false
}
cr.Enable(disableOthers)
```

Also, prefer shorter variable names for variables which have a very short scope.

File: pkg/minikube/cruntime/containerd.go [hunk 2]
```diff
@@ -79,9 +79,11 @@ func (r *Containerd) Available() error {
 }
 
 // Enable idempotently enables containerd on a host
-func (r *Containerd) Enable() error {
-	if err := disableOthers(r, r.Runner); err != nil {
-		glog.Warningf("disableOthers: %v", err)
+func (r *Containerd) Enable(disableOtherContainerEngines bool) error {
```
Comment (tstromberg) [inline pkg/minikube/cruntime/containerd.go]: Prefer shorter variable names for a short scope, such as `disableOthers`. Leave the long variable names for globals.

Comment (tstromberg) [reply pkg/minikube/cruntime/containerd.go]: Context: https://github.com/golang/go/wiki/CodeReviewComments#variable-names

Comment (marcosdiez) [reply pkg/minikube/cruntime/containerd.go]: Hi, although I agree, I can't do it. We have a function called `disableOthers` already.
I can't think about another good short name.
Do you have any suggestions ?

**Comment by marcosdiez** [comment]
Just for the record my build passed although Travis says otherwise: https://travis-ci.org/kubernetes/minikube/builds/549788904?utm_source=github_status&utm_medium=notification

**Comment by medyagh** [comment]
@minikube-bot OK to test

File: pkg/minikube/cruntime/crio.go [hunk 3]
```diff
@@ -78,9 +78,11 @@ func (r *CRIO) Active() bool {
 }
 
 // Enable idempotently enables CRIO on a host
-func (r *CRIO) Enable() error {
-	if err := disableOthers(r, r.Runner); err != nil {
-		glog.Warningf("disableOthers: %v", err)
+func (r *CRIO) Enable(disableOtherContainerEngines bool) error {
```
Comment (medyagh) [inline pkg/minikube/cruntime/crio.go]: lets change  `disableOtherContainerEngines` to `disOthers`

**Comment by RA489** [comment]
@minikube-bot OK to test

**Pull request merged**